### PR TITLE
Improve function separation in ffc-pay-batch-processor

### PR DESCRIPTION
**Pull request by steve-dickinson**
https://eaflood.atlassian.net/browse/SFI-1921

### CODE REVIEW

**Review by johnwatson484 — COMMENTED** [review]
Couple of refactoring suggestions added.

Also failing to deploy to Kubernetes for PR deployment, but I think that was due to the testing of the managed identity.  That's now sorted so re-running pipeline should hopefully succeed.

File: app/process-batches/process-payment-file.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  try {
+    console.log(`Processing ${filename}`)
+    const reprocessed = await reprocessIfNeeded(filename, schemeType)
+
+    if (!reprocessed) {
+      await checkSequence(schemeType, filename)
+    }
+  } catch (err) {
+    console.error(`Error thrown processing ${filename}`)
+    console.error(err)
```
[the remark this answers is not in the document]
Comment (steve-dickinson) [reply app/process-batches/process-payment-file.js:16]: The event would have no correlationalId to link to any payment request as its before the processing, but it should be logged as part of app insights? Maybe we need to look at raising some on the errors logged as alerts in Azure Monitor.

File: app/process-batches/process-payment-file.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+const downloadAndParse = require('./download-and-parse')
+const { sendBatchErrorEvent } = require('../event')
+
+async function processPaymentFile (filename, schemeType) {
+  try {
+    console.log(`Processing ${filename}`)
+    const reprocessed = await reprocessIfNeeded(filename, schemeType)
+
+    if (!reprocessed) {
+      await checkSequenceAndPerfomAction(schemeType, filename)
```
Comment (johnwatson484) [inline app/process-batches/process-payment-file.js:13]: Still think the name can be approved.  `PerformAction` is a bit generic and doesn't really describe what is going to happen.  Also don't think `checkSequence` needs to be in the name as that's just a sub action of the function being called.  The calling function doesn't need to care about the details of that action.

The action is analysing the file and then either parsing it, ignoring it or quarantining it.  Not sure how best to describe that activity

**Review by johnwatson484 — DISMISSED** [review]

**Review by johnwatson484 — APPROVED** [review]
LGTM

**Pull request merged**